### PR TITLE
Handle unit of '*' from NDS2

### PR DESCRIPTION
**Pull request by areeda**
Detect units of '*' and override so we can calculate spectra

### CODE REVIEW

**Comment by duncanmmacleod** [comment]
@areeda, thanks for the PR. The CI is failing because other tests (unrelated to this change) are out-of-date. Can you rebase your branch against the current `upstream/develop`?

```
git fetch upstream/develop
git rebase -i upstream/develop
```

**Comment by duncanmmacleod** [comment]
@areeda, what's the relevance of `agg` to this PR?

**Comment by areeda** [comment]
Sorry reused a branch to debug, such a small change I lazily didn't create a new one.
I can do that iif you like.

**Comment by areeda** [comment]
Are those CI failures significant?

**Comment by duncanmmacleod** [comment]
@areeda, the CI failures are obscuring the impact of the new additions, so if you can rebase as I posted then `git push --force` the CI should rerun and we might be able to see a bit better.

File: gwpy/cli/cliproduct.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
 
 from dateutil.parser import parser
 
-from ..timeseries import TimeSeries
+from gwpy.timeseries import TimeSeries
```
Comment (duncanmmacleod) [inline gwpy/cli/cliproduct.py]: @areeda, what's the reason for this change?

File: gwpy/cli/cliproduct.py [hunk 2]
```diff
@@ -801,13 +803,14 @@ def annotate_save_plot(self, args):
         all_units = set()
         for ts in self.timeseries:
             un = str(ts.unit)
-            if (un != 'undef') & (un != ''):
+            if (un != 'undef') & (un != '') & (un != '*'):
```
Comment (duncanmmacleod) [inline gwpy/cli/cliproduct.py]: it might be cleaner to set up a list in the module header that contains all of the proscribed units,

```python
BAD_UNITS = {'undef', '', '*'}
```

then just do something like

```python
            if str(ts.unit) in BAD_UNITS:
                ts.override_unit('')
            else:
                all_units.add(str(ts.unit))

File: gwpy/cli/cliproduct.py (hunk 1 above)
Comment (areeda) [reply gwpy/cli/cliproduct.py]: I think it's a tad more obvious plusI can copy/pasteinto ipython
No big deal I can reverse it if you likr

File: gwpy/cli/cliproduct.py (hunk 2 above)
Comment (areeda) [reply gwpy/cli/cliproduct.py]: I've been wondering whether they should be handled at this level at all? It was originally put in there to avoid confusion with the transition. The only real problem is when we do arithmetic on the units and for the most part that is being handled in the timeseries object I believe.

I like your suggestion and I will make the change.

File: gwpy/cli/cliproduct.py (hunk 1 above)
Comment (duncanmmacleod) [reply gwpy/cli/cliproduct.py]: yes, please revert. since this module is part of `gwpy`, it should use relative imports for relative objects as much as possible.

File: gwpy/cli/cliproduct.py [hunk 3]
```diff
@@ -801,13 +803,14 @@ def annotate_save_plot(self, args):
...
                 all_units.add(un)
             else:
+                ts.override_unit('')
                 all_units.add('Counts')
 
         if len(all_units) == 1:
-            self.units = all_units.pop()
+            self.units = label_to_latex(all_units.pop())
         else:
             self.units = 'Counts'
```
Comment (duncanmmacleod) [inline gwpy/cli/cliproduct.py]: @areeda, since 'Counts' is an actual physical unit, I don't think we should use it as the default. Perhaps something like

```python
        if len(all_units) == 1:
            self.units = label_to_latex(all_units.pop())
        elif len(all_units) > 1:
            self.units = 'multiple units'
        else:
            self.units = 'undef'
```

File: gwpy/cli/cliproduct.py (hunk 2 above)
Comment (duncanmmacleod) [reply gwpy/cli/cliproduct.py]: @areeda, the units parser knows about `'undef'`, and `''` is a legitimate way of specifying 'dimensionless', so please only include `'*'` in your `BAD_UNITS` `set`. We might consider handling that at the unit parser level, as you say.

File: gwpy/cli/cliproduct.py [hunk 4]
```diff
@@ -58,6 +58,7 @@ class CliProduct(object):
     """
 
     __metaclass__ = abc.ABCMeta
+    BAD_UNITS = ['undef', '', '*']
```
Comment (duncanmmacleod) [inline gwpy/cli/cliproduct.py]: @areeda, can you change this to just

```python
    BAD_UNITS = {'*',}
```

since the other two are legitimate unit choices? (also, `a in {set}` is slightly more efficient than `a in {list}` (but only really for large lists).

Comment (areeda) [reply gwpy/cli/cliproduct.py]: these are in latest push
